### PR TITLE
Use ubi instead of golang image for main Dockerbuild

### PR DESCRIPTION
**Pull request by idlewis**
To deal with dockerhub rate limits 

See WASdev/websphere-liberty-operator#270

### CODE REVIEW

**Review by halim-lee — APPROVED** [review]
Looks good!

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Build the manager binary
-FROM golang:1.20 as builder
+FROM registry.access.redhat.com/ubi8 as builder
```
Comment (leochr) [inline Dockerfile:2]: Can `registry.access.redhat.com/ubi8/ubi-minimal:latest` be used instead?

**Review by leochr — COMMENTED** [review]
@idlewis Wondering whether the operator binary from the new approach be compared with the current approach to ensure there isn't a significant difference

File: Dockerfile (hunk 1 above)
Comment (idlewis) [reply Dockerfile]: @leochr I've pushed some updates to use ubi-minimal.
I've tested locally and have a pipeline build running now.

**Comment by idlewis** [comment]
> @idlewis Wondering whether the operator binary from the new approach be compared with the current approach to ensure there isn't a significant difference

@leochr I have had a look around and I can't find a method to do a meaningful compare.
`go version -m <go binary>`
lists the go version and modules used to compile a binary, that is the best I have found.

One issue the above command did show is that I was using go 1.20.0 rather than 1.20.10. It seems the golang 1.20 docker image uses the 'latest' 1.20 version of go. However, the go download URL for 1.20 seems to specifically get 1.20.0. I haven't found an alternative download URL. This would mean we would have to update the docker file regularly to keep up with minor version updates.

File: Dockerfile [hunk 2]
```diff
@@ -1,5 +1,9 @@
 # Build the manager binary
-FROM golang:1.20 as builder
+FROM registry.access.redhat.com/ubi8-minimal:latest as builder
+ARG GO_PLATFORM=amd64
+env PATH=$PATH:/usr/local/go/bin
```
Comment (leochr) [inline Dockerfile:4]: Should this be `ENV` instead of `env`?

File: Dockerfile [hunk 3]
```diff
@@ -1,5 +1,9 @@
...
+RUN microdnf install tar gzip
+RUN curl -L --output - "https://golang.org/dl/go1.20.10.linux-${GO_PLATFORM}.tar.gz" | tar -xz -C /usr/local/
```
Comment (leochr) [inline Dockerfile:6]: The Go version should be parameterized. Link it to the OnePipeline's config `go-version`

https://github.ibm.com/websphere/operators/blob/3f38eb6677e355cdc7a5320a0c2cb9690709301f/scripts/pipeline/containerize-stage.sh#L109

**Review by leochr — COMMENTED** [review]
@idlewis Thanks for the PR. Added a few comments.

**Comment by leochr** [comment]
@idlewis The instructions at https://go.dev/doc/install state to use 
`rm -rf /usr/local/go && tar -C /usr/local -xzf go1.21.3.linux-amd64.tar.gz`

It doesn't seem like UBI image includes `/usr/local/go` but it would be good to remove the directory.
Also wondering if the `-f` option should be used on `tar` command

File: Dockerfile (hunk 2 above)
Comment (idlewis) [reply Dockerfile:5]: Yes. It seems to work with 'env' but that isnt' documented, so I've updated it.

**Comment by idlewis** [comment]
> @idlewis The instructions at https://go.dev/doc/install state to use `rm -rf /usr/local/go && tar -C /usr/local -xzf go1.21.3.linux-amd64.tar.gz`
> 
> It doesn't seem like UBI image includes `/usr/local/go` but it would be good to remove the directory. Also wondering if the `-f` option should be used on `tar` command

I've added the `rm` command.
The `-f` isn't needed in the tar command as we are extracting from stdin, instead of a file

**Review by leochr — APPROVED** [review]
@idlewis Looks good. Thank you

**Pull request merged**